transient-aws/hive-example/dispatch.sh

```diff
@@ -0,0 +1,316 @@
+    shift # past argument=value
+    ;;
+    -n=*|--ssh-username=*)
+    SSH_USERNAME="${i#*=}"
+    shift # past argument=value
+    ;;
+    -i=*|--ssh-private-key=*)
+    SSH_PRIVATE_KEY="${i#*=}"
+    shift # past argument=value
+    ;;
+    -f=*|--provision-config=*)
+    PROVISION_CONFIG="${i#*=}"
+    shift # past argument=value
+    ;;
+    -t|--terminate)
+    TERMINATE="true"
+    shift
+    ;;
+    -h|--help)
+    usage
+    exit 0
+    shift
+    ;;
+    *)
+          # unknown option
+    ;;
+esac
+done
+
+CLUSTER_URL="${DIRECTOR_SERVER}/api/v4/environments/${ENVIRONMENT_NAME}/deployments/${DEPLOYMENT_NAME}/clusters/${CLUSTER_NAME}"
+
+WGET="wget -O - --content-on-error --no-verbose --user ${DIRECTOR_USER} --password ${DIRECTOR_PASSWORD} --auth-no-challenge"
+
+terminate() {
+    for i in {5..0}; do
+        echo -ne "Terminating cluster ${CLUSTER_NAME} in $i seconds"'\r'
+        sleep 1
+    done
+
+    #${WGET} --method DELETE "${CLUSTER_URL}"
+    # Standard Centos version of wget doesn't support --method flag
+
+    curl -u ${DIRECTOR_USER}:${DIRECTOR_PASSWORD} -X "DELETE" "${CLUSTER_URL}"
+
+
+    STAGE=$(${WGET} -q "${CLUSTER_URL}/status" | jq -r ".stage");
+    if [[ "${STAGE}" != "TERMINATING" ]]; then
+        echo "Unable to terminate cluster. Stage is ${STAGE}."
+        echo "Please terminate cluster ${CLUSTER_NAME} manually!"
+        exit -1
+    fi
+
+    while
+        STAGE=$(${WGET} -q "${CLUSTER_URL}/status" | jq -r ".stage");
+        [ "${STAGE}" == "TERMINATING " ];
+    do
+        echo "Cluster ${CLUSTER_NAME} is terminating. Waiting for 10 seconds ..."
+        sleep 10
+    done
+
+    echo "Cluster ${CLUSTER_NAME} is terminated"
+
+    if [[ "${KEEP_CM}" == "false" ]]; then
+        ENV_URL="${DIRECTOR_SERVER}/api/v4/environments/${ENVIRONMENT_NAME}/deployments/${DEPLOYMENT_NAME}/"
+        echo "Terminating CM Instance"
+
+         #${WGET} --method DELETE "${ENV_URL}"
+        curl -u ${DIRECTOR_USER}:${DIRECTOR_PASSWORD} -X "DELETE" "${ENV_URL}"
+
+         while
+            STAGE=$(${WGET} -q "${ENV_URL}/status" | jq -r ".stage");
+            [ "${STAGE}" == "TERMINATING" ];
+        do
+            echo "Environment ${DEPLOYMENT_NAME} is terminating. Waiting 10 seconds"
+            sleep 10
+        done
+    fi
+    echo "Deployment ${DEPLOYMENT_NAME} is terminated"
+
+}
+
+#
+# Import and job dispatch logic
+#
+
+# Provision the cluster if not in job-submission mode
+PROVISIONED="false"
+if [[ -f "${PROVISION_CONFIG}" ]]; then
+    # Path to config file with all the details needed to create the cluster
+
+    # Import client config to Director
+    PROVISIONED="true"
+    echo "Importing client config ${PROVISION_CONFIG} to ${DIRECTOR_SERVER}"
+
+    ${WGET} --header="Content-Type: text/plain" --post-file "${PROVISION_CONFIG}" \
+        "${DIRECTOR_SERVER}/api/v4/import?clusterName=${CLUSTER_NAME}&deploymentName=${DEPLOYMENT_NAME}&environmentName=${ENVIRONMENT_NAME}"
```

Comment:
Similar to above - should the names be checked for safe use in URLs? Also, maybe encoding is required here.